In [1]:
n = int(input())

for i in range(0,n*2-1,1):
    for j in range(0, n*2-1,1):
        print(n - min(i,j, 2*n-1-1-i,2*n-1-1-j),end="")
        
    print()    


4
4444444
4333334
4322234
4321234
4322234
4333334
4444444


In [10]:
num = int(input())

#part 1
n=1
while n<=num:
    print(n,end=' ')
    n=n+1
print()

#part 2
n=1
adder=num*2
while n<(num/2):
    j=1
    while j<=num:
        Seq_Ele=j+(adder*n)
        print(Seq_Ele,end=' ')
        j=j+1
    print()
    n=n+1
    
#part 3    
while n<num:
    j=1
    k=num-n-1
    while j<=num:
        Seq_Ele=j+(num*((num-n)+k))
        print(Seq_Ele,end=' ')
        j=j+1
    print()
    n=n+1
    k=k+1

5
1 2 3 4 5 
11 12 13 14 15 
21 22 23 24 25 
16 17 18 19 20 
6 7 8 9 10 


In [9]:
n=int(input())

n1=int((n+1)/2)
n2=int(n/2)
#part 1
for i in range(1,n1+1,1):
    
    #spaces
    for j in range(1,n1-i+1,1):
        print(" ",end='')
    #stars
    for k in range(1,2*i,1):
        print('*',end='')
    print()    

#part 2
for i in range(n2,0,-1):
    #spaces 
    spaces =1 
    l=n2-i+1
    while spaces<=l:
        print(" ",end='')
        spaces=spaces+1
    #stars
    stars =1
    k=2*i-1
    while stars<=k:
        print("*",end='')
        stars=stars+1
    print()  

7
   *
  ***
 *****
*******
 *****
  ***
   *


In [8]:
n = int(input())

for a in range(1,n+1,1):
    for b in range(0,a-1,1):
        print(" ",end='')
    for c in range(a,n+1,1):
        print(c,end='') 
    print()

for i in range(n-1,0,-1):
    for j in range(1,i,1):
        print(' ',end='')
    for k in range(i,n+1,1):
        print(k,end="")
    print()

6
123456
 23456
  3456
   456
    56
     6
    56
   456
  3456
 23456
123456


In [6]:
n = int(input())
i = 1

while i <= n:
    j=1
    while j <= n-i+1:
        if i%2==0:
            print('0',end='')
        else:
            print('1',end='')
        
        j=j+1
    print()    
    i = i + 1

5
11111
0000
111
00
1


In [24]:
n = int(input())
n1=int((n+1)/2)
n2=int(n/2)

#upper half
i=1
while(i<=n1):
    
    spaces=1
    while spaces<=i-1:
        print(' ',end="")
        spaces=spaces+1
    
    stars=1
    while stars<=i:
        print('* ',end="")
        stars=stars+1
    
    print()
    i=i+1  
    
#lower half
n2=int(n/2)
j=1
while j<=n2:
    spaces=1
    l=n2-j
    while spaces<=l:
        print(' ',end='')
        spaces=spaces+1
        
    stars=1
    m=n2-j+1
    while stars<=m:
        print('* ',end='')
        stars=stars+1
        
    print()
    j=j+1
        

11
* 
 * * 
  * * * 
   * * * * 
    * * * * * 
     * * * * * * 
    * * * * * 
   * * * * 
  * * * 
 * * 
* 


In [19]:
n = int(input())

i=1
k=1
t=1
while i<=n: 
    #part 1
    j=1
    while j<=n:
     
        if(j==k):
            print('*',end="")
        else:
            print('0', end="")
        
        j=j+1      
    print('*',end='')
    k=k+1    
    
    #part 2
    m=n
    while m>=1:
        if(m==t):
            print("*",end='')
        else:
            print("0",end="")
        m=m-1
            
    t=t+1
    i=i+1
    print()
   
    

5
*0000*0000*
0*000*000*0
00*00*00*00
000*0*0*000
0000***0000


In [4]:
n = int(input())
i = 1

while i <= n:
 
    #part 1 - spaces
    spaces = 1
    while spaces <= n-i:
        print(' ', end='')
        spaces = spaces + 1
    
    #part 2 - decreasing seq.
    p = i
    while p >= 1:
        print(p, end='')
        p = p -1
        
    #part 3 - increasing seq.
    j = 2
    while j <= i:
        print(j, end="")
        j = j + 1
        
    print()
    i = i + 1

5
    1
   212
  32123
 4321234
543212345


In [3]:
n = int(input())
i = 1

while i <= n:
    #part 1 - increasing seq.
    j = 1
    while j <= i:
        print(j, end="")
        j = j + 1
    
    #part 2 - spaces
    spaces = 1
    s= 2*n-2*i
    while spaces <= s:
        print(' ', end='')
        spaces = spaces + 1
    
    #part 3 - decreasing seq.
    p = i
    while p >= 1:
        print(p, end='')
        p = p -1
        
    print()
    i = i + 1

4
1      1
12    21
123  321
12344321


In [2]:
n = int(input())
i = 1
while i <= n:
    spaces = 1
    # spaces
    while spaces <= n - i:
        print(' ', end='')
        spaces = spaces + 1
    p = 1
    j = 1
    # increasing seq
    while j <= i:
        print(p, end='')
        j = j + 1
        p = p + 1
    # decreasing seq
    p = i - 1
    while p >= 1:
        print(p, end='')
        p = p -1
    print()
    i = i + 1

4
   1
  121
 12321
1234321


In [6]:
n = int(input())
i = 1
while i <= n:
    j = 1
    charP = chr(ord('A') + i - 1)
    while j <= i:
        print(charP, end="")
        j = j + 1
    print()
    i = i + 1   

4
A
BB
CCC
DDDD


In [9]:
n = int(input())
i = 1
while i <= n:
    j = 1
    while j <= i:
        charP = chr(ord('A') + j - 1)
        print(charP, end="")
        j = j + 1
    print()
    i = i + 1   

4
A
AB
ABC
ABCD


In [10]:
n = int(input())
i = 1
while i <= n:
    j = 1
    while j <= n:
        charP = chr(ord('A') + j - 1)
        print(charP, end="")
        j = j + 1
    print()
    i = i + 1   

4
ABCD
ABCD
ABCD
ABCD


In [4]:
n = int(input())

i=1
print('1')
while i<n:
    
    print('1', end='')
    j=1
    while j<i:
        print('2',end='')   
        j=j+1
    print('1', end='')
    print()
    i = i+1

4
1
11
121
1221


In [3]:
n = int(input())
i=1

print('1')
while i<n:
    print(i, end='')
    
    j=1
    while j<i:
        print('0',end='')   
        j=j+1
    
    print(i, end='')
    print()
    i = i+1

4
1
11
202
3003


In [1]:
n = int(input())
i = 1
p = 1
while i <= n:
    j = 1
    while j <= i:
        print(p, end="")
        j = j + 1
    print()
    i = i + 1 

4
1
11
111
1111


In [2]:
n = int(input())
i = 1
while i <= n:
    j=1
    while j <= n-i+1:
        print(j,end = '')
        j=j+1
    print()    
    i = i + 1

4
1234
123
12
1


# Armstrong Number

In [5]:
# Function to calculate order of the number 
def order(n):
    temp = n
    times = 0
    while temp > 0:
        times += 1
        temp //= 10
    
    return times    
        
def checkArmstrong(num):
    sum = 0
    temp = num
    o = order(num)
    
    while temp > 0:
        digit = temp % 10
        sum = sum + (digit ** o)
        temp = temp // 10
        
    if num == sum:
        return True
    else:
        return False
        
num = int(input())
isArmstrong = checkArmstrong(num)
if(isArmstrong):
	print('true')
else:
	print('false')

1634
true


## Prime Number

In [8]:
n = int(input())
d = 2
flag = False
while d < n:
    if (n % d == 0):
        flag = True
    d = d + 1

if flag:
    print("not prime")
else:
    print("prime")

4
not prime


In [1]:
#Prime numbers from 2 to N
n = int(input())
k = 2
while k <= n:
    d = 2
    flag = False
    while d < k:
        if (k % d == 0):
            flag = True
        d = d + 1
    if not(flag):
        print(k)
    k = k + 1

10
2
3
5
7


## Check whether the number N is the member of fibonacci series or not!
### 0,1,1,2,3,5,8,13,21

In [2]:
N = int(input())
a = 0
b = 1

while a < N:
    print(a)
    c = a + b
    a = b
    b = c
if a == N:
    print("It is a member of fibonacci series")
else:
    print("Not a member of fibonacci series")


8
0
1
1
2
3
5
It is a member of fibonacci series


## Nth Fibonacci number

Nth term of fibonacci series F(n) is calculated using following formula -
    F(n) = F(n-1) + F(n-2), 
Provided N you have to find out the Nth Fibonacci Number. Also F(1) = F(2) = 1.

Fibonacci numbers are defined by the sequence f(0) = 0, f(1) = 1 and f(n) = f(n – 1) + f(n – 2) for n >= 2. The program prompts the user to enter n and it prints the nth Fibonacci number.

In [4]:
def fibonacci(n):
    if n <= 1:
        return n
    else:
        return fibonacci(n-1) + fibonacci(n-2)

print(fibonacci(int(input())))

4
3


### Second approach of Nth term of fibonacci series

In [5]:
n = int(input())
c = 0
a = 0
b = 1

while c < n-1:
    nth = a + b
    a = b
    b = nth
    c = c + 1
print(nth)    

1
1


## Sum of even & odd digits of a number

Write a program to input an integer N and print the sum of all its even digits and sum of all its odd digits separately.
Digits means numbers not the places. That is, if the given integer is "13245", even digits are 2 & 4 and odd digits are 1, 3 & 5.

**Algorithm**

1.Get the number
2.Declare a variable to store the sum and set it to 0
3.Repeat the next two steps till the number is not 0
4.Get the rightmost digit of the number with help of remainder ‘%’ operator by dividing it with 10 and add it to sum.
5.Divide the number by 10 with help of ‘/’ operator
6.Print or return the sum

In [1]:
n = int(input())
soe = 0 
sod = 0

while n!=0:
    dig=n%10
    if dig%2 == 0:
        soe = soe+dig
    else:
        sod = sod+dig
    n = int(n/10)
print(soe, " ", sod)

1234
6   4


## Palindrome number

Write a program to determine if given number is palindrome or not. Print true if it is palindrome, false otherwise.
Palindrome are the numbers for which reverse is exactly same as the original one. For eg. 121

**Logic**
1.	User must first enter the value of the integer and store it in a variable.
2.	The value of the integer is then stored in another temporary variable.
3.	The while loop is used and the last digit of the number is obtained by using the modulus operator.
4.	The last digit is then stored at the one’s place, second last at the ten’s place and so on.
5.	The last digit is then removed by truly dividing the number with 10.
6.	This loop terminates when the value of the number is 0.
7.	The reverse of the number is then compared with the integer value stored in the temporary variable.
8.	If both are equal, the number is a palindrome.
9.	If both aren’t equal, the number isn’t a palindrome.
10.	The final result is then printed.

In [8]:
n=int(input())
temp=n
rev=0
while(n>0):
    dig=n%10
    rev=rev*10+dig
    n=n//10
if(temp==rev):
    print("true")
else:
    print("false")

121
true


## Fahrenheit to Celsius
Given three values - Start Fahrenheit Value (S), End Fahrenheit value (E) and Step Size (W), you need to convert all Fahrenheit values from Start to End at the gap of W, into their corresponding Celsius values and print the table.

In [1]:
S = int(input())
E = int(input())
W = int(input())

while S <= E: 
    C = int((S - 32) * 5/9)
    print(S ,"\t", C)
    S = S + W

0
100
20
0 	 -17
20 	 -6
40 	 4
60 	 15
80 	 26
100 	 37


## Calculator
Write a program that works as a simple calculator. It reads an integer for choice.
1. If the choice is 1, 2 integers are taken for input and their sum is printed.
2. If the choice is 2, 2 integers are taken for input and their difference is printed.
3. If  the choice is 3, 2 integers are taken for input and their product is printed.
4. If  the choice is 4, 2 integers are taken for input and their quotient is printed.
5. If  the choice is 5, 2 integers are taken for input and their remainder is printed.
6. If the choice is 6, the program exits, 
For any other choice, print "Invalid Operation" and ask for choice again.

In [3]:
# Write your code here
flag=True

while flag:
    
    choice = int(input())
    if choice==1:
        num1 = int(input())
        num2 = int(input())
        print(int(num1+num2))
        
    elif choice==2:
        num1 = int(input())
        num2 = int(input())
        print(int(num1-num2))
    
    elif choice==3:
        num1 = int(input())
        num2 = int(input())
        print(int(num1*num2))
        
    elif choice==4:
        num1 = int(input())
        num2 = int(input())
        print(int(num1/num2))

    elif choice==5:
        num1 = int(input())
        num2 = int(input())
        print(int(num1%num2))
    
    elif choice==6:
        flag = False
        
    else:
        print("Invalid Operation")

1
4
5
9
2
2
1
1
6


## Reverse of a number
Write a program to generate the reverse of a given number N. Print the corresponding reverse number.

In [4]:

r = 0
n = int(input())    
while n>0:    
    rem = int(n%10)    
    r = r*10 + rem   
    n = n//10      
    
print(r)    
    


1234
4321


## Print prime numbers from 2 to n

In [11]:
n = int(input())
k = 2
while k <= n:
    d = 2
    flag = False
    while d < k:
        if (k % d == 0):
            flag = True
            break
        d = d + 1
    if not(flag):
        print(k)
    k = k + 1

100
2
3
5
7
11
13
17
19
23
29
31
37
41
43
47
53
59
61
67
71
73
79
83
89
97


## Sum of Even Numbers

Given a number N, print sum of all even numbers from 1 to N.

In [4]:
n = int(input())
Sum = 0
i = 0

while i<=n :
    if i%2 == 0:
        Sum = Sum + i
        i = i + 1
    else:
        i = i + 1
   
print(Sum)

6
12


## Sum of n numbers

Given an integer n, find and print the sum of numbers from 1 to n.

In [5]:
n = int(input())
sum = 0
i = 1

while i<=n :
    sum = sum + i
    i = i + 1
   
print(sum)

10
55


## Check number

Given an integer n, find if n is positive, negative or 0.
If n is positive, print "Positive"
If n is negative, print "Negative"
And if n is equal to 0, print "Zero".

In [6]:
n = int(input())

if n>0 :
    print('Positive')
elif n<0 :
    print('Negative')
else :
    print('Zero')
    

10
Positive


## Rectangular Area

You are given a rectangle in a plane. The corner coordinates of this rectangle is provided to you. You have to print the amount of area of the plane covered by this rectangles.
The end coordinates are provided as four integral values: x1, y1, x2, y2. It is given that x1 < x2 and y1 < y2.
Input format:
The first line of input contains an integer x1 (1 <= x1 <= 10)
The second line of input contains an integer y1 (1 <= y1 <= 10) 
The third line of input contains an integer x2 (1 <= x2 <= 10)
The fourth line of input contains an integer y2 (1 <= y2 <= 10) 

In [7]:
x1 = int(input())
y1 = int(input())
x2= int(input())
y2 = int(input())

l = x2-x1
b = y2-y1

print(l*b)

1
1
3
3
4


## Simple Interest

In [5]:
p = 100
r = 12
t = 2
# si = p * r * t // 100
#si = t//100 * r * p
si = (p * r * t)//100
si

24